resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -99,6 +102,17 @@ private[spark] class ExecutorPodsAllocator(
   @volatile private var deletedExecutorIds = Set.empty[Long]
 
   def start(applicationId: String, schedulerBackend: KubernetesClusterSchedulerBackend): Unit = {
+    // Wait until the driver pod is ready before starting executors, as the headless service won't
+    // be resolvable by DNS until the driver pod is ready.
+    try {
+      kubernetesClient.pods()
+        .withName(kubernetesDriverPodName.get)
+        .waitUntilReady(driverPodReadinessTimeout, TimeUnit.MINUTES)
+    } catch {
+      case e: InterruptedException =>
+        logWarning(s"Timeout waiting for driver pod ${kubernetesDriverPodName.get} get ready in " +
+          s"namespace $namespace")
+    }
```

Comment:
If we are going to just `logWarning`, the following code pattern will be better like the other places.
```scala
Utils.tryLogNonFatalError {
  kubernetesClient.pods()
    .withName(kubernetesDriverPodName.get)
    .waitUntilReady(driverPodReadinessTimeout, TimeUnit.MINUTES)
}
```

Reply:
If we just do `logWarning` and continue to create executor pods after this timeout, it's very likely executor pods will still hit the `UnknownHostException` issue.
Considering this, throwing a SparkException may be more intuitive and fail-fast. Please let me know your thoughts on this. Thanks.
Like here:
https://github.com/apache/spark/blob/0342dcb6289bae67f6ab742b8fd03b2d653e52ea/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala#L73-L79

Reply:
-1 for `SparkException` idea. That is a huge regression.

`UnknownHostException` is not a job blocker because Spark is going to create the required executors in any way eventually. If you throws SparkException, you change will be a breaker of the existing Spark pipelines indeed.

Reply:
As the [JIRA issue](https://issues.apache.org/jira/browse/SPARK-32975) describes, driver will not create new executors when this error happens.
> Once this error happens, driver doesn't restart executor.

However, other executors may be running fine. It's just that the number of healthy executors will be less than expected.    So I agree not to use `SparkException` here.

Reply:
I don't think the error report is correct because Apache Spark doesn't use `restart` executor at all. The driver will keep creating new executors because the executors with `UnknownHostException` will be dead on the failure. I'm currently describing the Apache Spark behavior, not 3rd party operator behavior.